### PR TITLE
ENH: Default section for Card.add_* methods

### PR DESCRIPTION
**Pull request by BenjaminBossan**
## Description

The `add_*` methods for `Card` now have default sections (the one used in the skops template) and no longer set descriptions by default. `None` is no longer a valid argument for section.

This is useful because it allows us to completely remove all the template checks in the diverse `add_*` methods. Before this change, many `add_*` methods had to check the template and if it was a custom template, would raise an error when the section is not indicated.

All methods have been homogenized in that they no longer add a description by default and put a placeholder there if empty. This means that after this PR is merged, the same code will sometimes produce a different looking model card.

## Comment

For the `Card` class, this is a nice simplification of the code. The main change affects the unit tests. All tests that used to raise when using a custom template without a section argument no longer raise but use the default template. Furthermore, since some default descriptions have been removed, other tests had to be touched too. Apart from that, some minor cleanups in the tests.

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
@skops-dev/maintainers ready for review

**Review by adrinjalali — APPROVED** [review]
Love this, merging the latest `main` to double check before merging.

**Pull request merged**
